index.d.ts

```diff
@@ -61,7 +62,7 @@ declare module 'fastify' {
 
   interface FastifyRequest {
     auth0Verify: Auth0Verify
-    jwtDecode: typeof decode
+    jwtDecode: (options?: DecodeOptions & { complete: true } | DecodeOptions & { json: true } | DecodeOptions) => null | JwtPayload | Jwt | string
```

Comment:
can this be simplified to

```suggestion
    jwtDecode: (options?: DecodeOptions & { complete: true, json: true }) => null | JwtPayload | Jwt | string
```

Reply:
can you apply it and see if the build is green? if it is, we can go ahead and merge and do a new release of this package

Reply:
I'll first try and change it locally as I see the test failing.

Reply:
So actually it **can't** be done like suggested. As per the TS documentation (https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types) using the `&` operator will mark the type as the intersection of the (two) types. In our case the suggestion would translate to something like this:
```
{ complete?: boolean | undefined, json?: boolean | undefined } & { complete: true, json: true }
```
The TS compiler will complain in this case, as both properties are incompatible across these two types.

So we are left with two options:
1. we keep it as it is, reflecting the types from the `jsonwebtoken` library
2. we introduce a shorter type, like this: `options?: DecodeOptions | { complete: true, json: true }` (notice that I'm using the `|` instead of the `&` )

What do you think?